app/src/main/java/com/eis/phoneringer/AppManager.java

```diff
@@ -0,0 +1,96 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+import android.media.Ringtone;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.widget.Toast;
+
+import androidx.annotation.RequiresApi;
+
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+
+/**
+ * This is a singleton class used to manage a received RingCommand or to send one
+ *
+ * @author Alberto Ursino, Luca Crema, Alessandra Tonin, Marco Mariotto
+ */
+public class AppManager {
+
+    private static final String WRONG_PASSWORD_TOAST = "Wrong password";
+    private static final int TIMEOUT_TIME = 30 * 1000; //30 seconds
+
+    /**
+     * Instance of the class that is instantiated in getInstance method
+     */
+    private static AppManager instance = null;
+
+    /**
+     * Private constructor
+     */
+    private AppManager() {
+    }
+
+    /**
+     * @return the AppManager instance
+     */
+    public static AppManager getInstance() {
+        if (instance == null)
+            instance = new AppManager();
+        return instance;
+    }
+
+    /**
+     * If the password of the message received is valid then play ringtone for fixed amount of time
+     *
+     * @param context     of the application
+     * @param ringCommand a ring command not null
+     */
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    public void onRingCommandReceived(Context context, RingCommand ringCommand, final Ringtone ringtone) {
```

Comment:
If you really really want the `ringCommand` to be not null you could add the @NonNull annotation. It's up to you, not necessary (you could also add it on context)